9.2/test/run

```diff
@@ -255,6 +255,66 @@ function run_tests() {
   fi
 }
 
+function run_replication_test() {
+  echo "Testing master-slave replication"
+  local cluster_args="-e POSTGRESQL_ADMIN_PASSWORD=pass -e POSTGRESQL_MASTER_USER=$PGUSER -e POSTGRESQL_MASTER_PASSWORD=$PASS"
+  local max_attempts=30
+
+  # Run the PostgreSQL master
+  docker run $cluster_args -p 5432  \
+    -d --cidfile ${CIDFILE_DIR}/master.cid $IMAGE_NAME run-postgresql-master >/dev/null
+  local master_ip
+  master_ip=$(get_container_ip master.cid)
+
+  # Run the PostgreSQL slave
+  docker run $cluster_args -e POSTGRESQL_MASTER_IP=${master_ip} \
+    -d --cidfile ${CIDFILE_DIR}/slave.cid $IMAGE_NAME run-postgresql-slave >/dev/null
+  local slave_ip
+  slave_ip=$(get_container_ip slave.cid)
+
+  # Check if the master knows about the slave
+  CONTAINER_IP=$master_ip
+  local i
+  for i in $(seq $max_attempts); do
+    result="$(postgresql_cmd -c "select client_addr from pg_stat_replication;" | grep "$slave_ip" || true)"
+    if [[ -n "${result}" ]]; then
+      echo "${slave_ip} successfully registered as SLAVE for ${master_ip}"
+      break
+    fi
+    if [[ "${i}" == "${max_attempts}" ]]; then
+      echo "The ${slave_ip} failed to register in MASTER"
+      echo "Dumping logs for $(get_cid slave.cid)"
+      docker logs $(get_cid slave.cid)
+      return 1
+    fi
+    sleep 1
+  done
+
+  local value
+  value=24
+  # Do some real work to test replication in practice
+  postgresql_cmd -c "CREATE TABLE t1 (a integer); INSERT INTO t1 VALUES ($value);"
+
+  CONTAINER_IP=$slave_ip
+  # Read value from slave and check whether it is expected
+  for i in $(seq $max_attempts); do
+    set +e
+    result="$(postgresql_cmd -c "select * from t1" | grep $value)"
+    set -e
+    if [[ ! -z "${result}" ]]; then
+      echo "${slave_ip} successfully got value from MASTER ${master_ip}"
+      break
+    fi
+    if [[ "${i}" == "${max_attempts}" ]]; then
```

Comment:
Ah, ok -- that is not infinite loop.  Then this condition is probably not needed at all.